src/marqo/tensor_search/models/settings_object.py

```diff
@@ -192,6 +192,7 @@
         NsFields.number_of_replicas: {
             "type": "integer",
             "minimum": 0,
+            "maximum": int(read_env_vars_and_defaults(EnvVars.MARQO_MAX_NUMBER_OF_REPLICAS)),
```

Comment:
you can use this method to parse env vars into ints: https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/tensor_search/utils.py#L202

It does things like give a useful error message if the value can't be parsed properly.